### ISSUE TITLE
Crash on Startup

### ISSUE THREAD

**Issue by daddydave-jpg** [Synthesis-Collective/SynthEBD#21]
huge fan of your zEBD, recently discovered your newest iteration. Downloaded, all prereqs met, crash every time:

SynthEBD has crashed with the following error:
Exception has been thrown by the target of an invocation.
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at System.Xaml.Schema.SafeReflectionInvoker.CreateInstanceCritical(Type type, Object[] arguments)
   at System.Xaml.Schema.XamlTypeInvoker.CreateInstance(Object[] arguments)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateInstanceWithCtor(XamlType xamlType, Object[] args)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateInstance(XamlType xamlType, Object[] args)
   at System.Xaml.XamlObjectWriter.Logic_CreateAndAssignToParentStart(ObjectWriterContext ctx)
   at System.Xaml.XamlObjectWriter.WriteStartMember(XamlMember property)
   at System.Xaml.XamlWriter.WriteNode(XamlReader reader)
   at System.Windows.Markup.WpfXamlLoader.TransformNodes(XamlReader xamlReader, XamlObjectWriter xamlWriter, Boolean onlyLoadOneNode, Boolean skipJournaledProperties, Boolean shouldPassLineNumberInfo, IXamlLineInfo xamlLineInfo, IXamlLineInfoConsumer xamlLineInfoConsumer, XamlContextStack`1 stack, IStyleConnector styleConnector)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at System.Windows.Application.LoadBamlStreamWithSyncInfo(Stream stream, ParserContext pc)
   at System.Windows.Application.LoadComponent(Uri resourceLocator, Boolean bSkipJournaledProperties)
   at System.Windows.Application.DoStartup()
   at System.Windows.Application.<.ctor>b__1_0(Object unused)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Object reference not set to an instance of an object.
   at SynthEBD.Settings_OBody.ImportBodySlides(HashSet`1 templateDescriptors) in D:\a\SynthEBD\SynthEBD\SynthEBD\Settings\Settings_OBody\Settings_OBody.cs:line 53
   at SynthEBD.MainWindow_ViewModel.LoadInitialSettingsViewModels() in D:\a\SynthEBD\SynthEBD\SynthEBD\MainWindow_ViewModel.cs:line 101
   at SynthEBD.MainWindow_ViewModel..ctor() in D:\a\SynthEBD\SynthEBD\SynthEBD\MainWindow_ViewModel.cs:line 56
   at SynthEBD.MainWindow..ctor() in D:\a\SynthEBD\SynthEBD\SynthEBD\MainWindow.xaml.cs:line 21
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)

Tried compatibility mode, launching as admin, launching from all 4 of my drives, from the directory SkyrimSE is in, from the modtool (vortex), I can't think of any other ways to T/S. Version 0.7.3

**Comment by daddydave-jpg** [comment]
Interestingly enough, I tested it out on my laptop (.Net 6.0.4) and it launched first try no issues straight from the download folder. Skyrim is not installed, and I got an environment prompt to select non default install location. Windows home laptop vs Windows Pro desktop, both fully updated on the same 21H2 22000.739 OS build. Lots of random shit installed on my Desktop from modding, creating, etc. but no malware/virus issues according to paid AVG, paid Norton, SFC scan, etc.

**Comment by daddydave-jpg** [comment]
Further troubleshooting, I noticed in my event log I had some issues with startup due to corrupt windows files. I went ahead and did a complete fresh install of windows 11. Bare bones, barely finished the "welcome to windows" BS and downloaded Vortex, reattached to my mods (stored on a different drive), downloaded SynthEBD, and .Net Framework 6.0.6 Still getting the exact same crash. I've narrowed it down to Ryzen (in the desktop) vs Intel (in the laptop). That's literally the only difference, besides Windows 11 home (laptop) and Windows 11 pro (desktop).

**Comment by Piranha91** [comment]
Sorry, I really should be commenting on these. I have just released the current version of the code as a new prerelease build (0.7.4). It might take a couple minutes after I post this comment for the .zip to become available. I have been really busy the past couple weeks, and will be for a few more days, so I have not been able to test this version to any reasonable extent, but enough people are having issues with the null reference on startup that I think this release is warranted. I will test more extensively when I get a chance, and then continue Noggog's suggested refactor to eliminate `PatcherEnvironmentProvider.Instance` and other suggested modifications. Thanks for your patience - in the meantime, please post if you run into any issues with this new release (and if not, let me know if I can close this issue).

**Comment by daddydave-jpg** [comment]
Will do good sir. Don't fret so much about being busy, you're providing a service for free! I fully recognize that, and I don't want to seem pushy or needy. I was simply trying to provide as much info as possible is all. I'll go ahead and give a go, report back here with any findings.

**Comment by daddydave-jpg** [comment]
Unfortunately, same issue. I tested it on my laptop as well and it works fine there. I'm attaching Sysinfo from both machines in an effort to provide as much info as possible. I just can't wrap my head around why it works on one machine and not the other when virtually everything is the same between them besides processors. Just to be clear, I'm running 6.0.6 .NET Framework on the desktop x64 and x86 both installed currently but I've tested having only one or the other installed as well. They are the .NET Desktop Runtime 6.0.6 straight from Microsoft.

SynthEBD has crashed with the following error:
Exception has been thrown by the target of an invocation.
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at System.Xaml.Schema.SafeReflectionInvoker.CreateInstanceCritical(Type type, Object[] arguments)
   at System.Xaml.Schema.XamlTypeInvoker.CreateInstance(Object[] arguments)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateInstanceWithCtor(XamlType xamlType, Object[] args)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.CreateInstance(XamlType xamlType, Object[] args)
   at System.Xaml.XamlObjectWriter.Logic_CreateAndAssignToParentStart(ObjectWriterContext ctx)
   at System.Xaml.XamlObjectWriter.WriteStartMember(XamlMember property)
   at System.Windows.Markup.WpfXamlLoader.TransformNodes(XamlReader xamlReader, XamlObjectWriter xamlWriter, Boolean onlyLoadOneNode, Boolean skipJournaledProperties, Boolean shouldPassLineNumberInfo, IXamlLineInfo xamlLineInfo, IXamlLineInfoConsumer xamlLineInfoConsumer, XamlContextStack`1 stack, IStyleConnector styleConnector)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at System.Windows.Application.LoadBamlStreamWithSyncInfo(Stream stream, ParserContext pc)
   at System.Windows.Application.LoadComponent(Uri resourceLocator, Boolean bSkipJournaledProperties)
   at System.Windows.Application.DoStartup()
   at System.Windows.Application.<.ctor>b__1_0(Object unused)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Object reference not set to an instance of an object.
   at SynthEBD.Settings_OBody.ImportBodySlides(HashSet`1 templateDescriptors) in D:\a\SynthEBD\SynthEBD\SynthEBD\Settings\Settings_OBody\Settings_OBody.cs:line 53
   at SynthEBD.SaveLoader.LoadInitialSettingsViewModels() in D:\a\SynthEBD\SynthEBD\SynthEBD\SaveLoader.cs:line 78
   at SynthEBD.MainWindow_ViewModel.Init() in D:\a\SynthEBD\SynthEBD\SynthEBD\MainWindow_ViewModel.cs:line 46
   at SynthEBD.MainWindow..ctor() in D:\a\SynthEBD\SynthEBD\SynthEBD\MainWindow.xaml.cs:line 28
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)

[Desktop](https://pastebin.com/E23fn7hT)
[Laptop](https://pastebin.com/3kPpXNrE)

Had to remove the error reporting from the laptop info as it was too long.
